src/object_detector/operations.jl

```diff
@@ -0,0 +1,229 @@
+import Statistics
+import Images
+
+
+
+""""
+nanmean()
+    computing the mean of an array, x, while neglegting NaN values.
+"""
+nanmean(x) = Statistics.mean(filter(!isnan,x))
+nanmean(x,y) = mapslices(nanmean,x,dims=y)
+
+""""
+nanstd()
+    computing the std of an array, x, while neglegting NaN values.
+"""
+nanstd(x) = Statistics.std(filter(!isnan,x))
+nanstd(x,y) = mapslices(nanstd,x,dims=y)
+
+
+
+@doc raw"""
+Add padding to array
+"""
+function add_padding(array::Matrix{T} where T<: Real, padding_width::Integer=7, pad_value::Real=P where P<: !isnan(P) )
```

Comment:
I do not understand thins function definition. The ` pad_value::Real=P where P<: !isnan(P)` part does not make any sense. You should not check the value of the function in the function header. That is for types and nan is not a type. Maybe what you want is 
```
function add_padding(array::Matrix{T},  padding_width::Integer=7, pad_value::Real ) where T<: Real
@assert !isnan(pad_value)
```